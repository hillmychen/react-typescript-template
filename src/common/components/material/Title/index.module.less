.title {
  line-height: 1;
  padding: 1px 0px;
  display: flex;
  align-items: center;
  .titleText {
    font-size: 16px;
    position: relative;
    margin-right: 1px;
    &.verticalHighlight {
      padding: 1px 8px;
    }
    &.verticalHighlight:before {
      content: '';
      position: absolute;
      width: 3px;
      top: 0;
      bottom: 0;
      left: 0;
      border-radius: 3px;
      background-color: var(--primary-color);
    }
  }
  .titleRight {
    margin-left: auto;
  }
  &.fullLine {
    padding: 8px 10px;
    border-radius: var(--border-radius-base);
    background-color: rgba(var(--table-header-bg), 0.8);
  }
}
