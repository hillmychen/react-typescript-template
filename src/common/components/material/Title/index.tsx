/*
 * @Author: Hughie
 * @Date: 2019-11-28 10:36:45
 * @Last Modified by: Hughie
 * @Last Modified time: 2019-11-28 10:41:33
 * 标题头，比如作为 PageCard 的 title 或者 FormBlock 的 title 等等
 */

import React from 'react'
import classnames from 'classnames'
import styles from './index.module.less'

export type TitleType = 'noStyle' | 'verticalHighlight' | 'fullLine' | 'highlightFullLine'

export interface TitleProps {
  text?: string | React.ReactNode
  className?: string
  /**
   * title 样式
   * @types
   * - noStyle 没有样式
   * - verticalHighlight(default) 左侧有一个垂直高亮竖线
   * - fullLine 占满整行
   * - highlightFullLine 左侧有一个垂直高亮竖线且占满整行
   */
  type?: TitleType
  right?: React.ReactNode
  left?: React.ReactNode
  style?: React.CSSProperties
}

export const Title: React.FC<TitleProps> = React.memo(props => (
  <div
    className={classnames(
      styles.title,
      {
        [styles.fullLine]: props.type === 'fullLine' || props.type === 'highlightFullLine',
      },
      props.className
    )}
    style={props.style}
  >
    {props.text && <strong className={classnames(styles.titleText, { [styles.verticalHighlight]: props.type === 'verticalHighlight' || props.type === 'highlightFullLine' })}>{props.text}</strong>}
    {props.left && <div>{props.left}</div>}
    {props.right && <div className={styles.titleRight}>{props.right}</div>}
  </div>
))

Title.defaultProps = {
  type: 'verticalHighlight',
}

export default Title
