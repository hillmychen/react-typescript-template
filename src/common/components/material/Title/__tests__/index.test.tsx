/*
 * @Author: Hughie
 * @Date: 2020-06-21 09:26:34
 * @LastEditTime: 2020-06-21 09:56:31
 * @LastEditors: Hughie
 * @Description:
 */
import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Title from '../index'

it('Title title', () => {
  const title = '标题'

  const { asFragment, getByText } = render(<Title text={title} />)
  expect(getByText(title)).not.toBeUndefined()
  expect(asFragment()).toMatchSnapshot()
})

it('Title noStyle', () => {
  const title = '标题'
  const { asFragment, getByText } = render(<Title text={title} type="noStyle" />)
  expect(getByText(title)).not.toBeUndefined()
  expect(asFragment()).toMatchSnapshot()
})

it('Title fullLine', () => {
  const title = '标题'
  const { asFragment } = render(<Title text={title} type="fullLine" />)
  expect(asFragment()).toMatchSnapshot()
})

it('Title fullLine', () => {
  const title = '标题'
  const { asFragment } = render(<Title text={title} type="verticalHighlight" />)
  expect(asFragment()).toMatchSnapshot()
})

it('Title className', () => {
  const { asFragment, container } = render(<Title text="标题" className="text" />)
  expect(container.firstChild).toHaveClass('text')
  expect(asFragment()).toMatchSnapshot()
})
