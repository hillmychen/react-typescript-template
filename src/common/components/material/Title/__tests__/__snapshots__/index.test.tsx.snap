// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title className 1`] = `
<DocumentFragment>
  <div
    class="title text"
  >
    <strong
      class="titleText verticalHighlight"
    >
      标题
    </strong>
  </div>
</DocumentFragment>
`;

exports[`Title fullLine 1`] = `
<DocumentFragment>
  <div
    class="title fullLine"
  >
    <strong
      class="titleText"
    >
      标题
    </strong>
  </div>
</DocumentFragment>
`;

exports[`Title fullLine 2`] = `
<DocumentFragment>
  <div
    class="title"
  >
    <strong
      class="titleText verticalHighlight"
    >
      标题
    </strong>
  </div>
</DocumentFragment>
`;

exports[`Title noStyle 1`] = `
<DocumentFragment>
  <div
    class="title"
  >
    <strong
      class="titleText"
    >
      标题
    </strong>
  </div>
</DocumentFragment>
`;

exports[`Title title 1`] = `
<DocumentFragment>
  <div
    class="title"
  >
    <strong
      class="titleText verticalHighlight"
    >
      标题
    </strong>
  </div>
</DocumentFragment>
`;
